<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Joseph Rickert">
<meta name="dcterms.date" content="2024-12-13">
<meta name="description" content="An attempt to capture the depth and breadth of what’s new on CRAN.">

<title>November 2024 Top 40 New CRAN Packages</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="2024NovTop40_files/libs/clipboard/clipboard.min.js"></script>
<script src="2024NovTop40_files/libs/quarto-html/quarto.js"></script>
<script src="2024NovTop40_files/libs/quarto-html/popper.min.js"></script>
<script src="2024NovTop40_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="2024NovTop40_files/libs/quarto-html/anchor.min.js"></script>
<link href="2024NovTop40_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2024NovTop40_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="2024NovTop40_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="2024NovTop40_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="2024NovTop40_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">November 2024 Top 40 New CRAN Packages</h1>
  <div class="quarto-categories">
    <div class="quarto-category">Top 40</div>
  </div>
  </div>

<div>
  <div class="description">
    An attempt to capture the depth and breadth of what’s new on CRAN.
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Joseph Rickert </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 13, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<p>One hundred tinny-four new packages made CRAN’s final cut in November. Here are my Top 40 picks in nineteen categories: Botany, Climate Analysis, Computational Methods, Data, Ecology, Education, Environment, Epidemiology, Genomics, Health Technology Assessment, Machine Learning, Mathematics, Medicine, Networks, Pharma, Statistics, Time Series, Utilities, and Visualization.</p>
<div class="columns">
<div class="column" style="width:45%;">
<section id="botany" class="level3">
<h3 class="anchored" data-anchor-id="botany">Botany</h3>
<p><a href="https://cran.r-project.org/package=MultiTraits">MultiTraits</a> v0.1.0: Implements analytical methods for multidimensional plant traits, including Competitors-Stress tolerators-Ruderals strategy analysis using leaf traits, Leaf-Height-Seed strategy analysis, Niche Periodicity Table analysis, and Trait Network analysis. See <a href="https://www.nature.com/articles/250026a0">Grime (1974)</a>, <a href="https://besjournals.onlinelibrary.wiley.com/doi/10.1111/1365-2435.12882">Pierce et al.&nbsp;(2017)</a> and other references provided for methods and the <a href="https://cran.r-project.org/web/packages/MultiTraits/vignettes/MultiTraits_tutorial.html">vignette</a> for a tutorial.</p>
<p><img src="MultiTraits.png" class="img-fluid" alt="Diagram of LHS Strategy"></p>
</section>
<section id="climate-analysis" class="level3">
<h3 class="anchored" data-anchor-id="climate-analysis">Climate Analysis</h3>
<p><a href="https://cran.r-project.org/package=chooseGCM">chooseGCM</a> v1.0.1: Implements methods to help selecting General <a href="https://en.wikipedia.org/wiki/General_circulation_model">Circulation Models (GCMs)</a> for projecting models to future scenarios. Functions include clustering algorithms, distance and correlation metrics, and a tailor-made algorithm to detect the optimum subset of GCMs that recreate the environment of all GCMs. Look <a href="">here</a> for examples.</p>
<p><img src="chooseGCM.png" class="img-fluid" alt="Various clustering plots"></p>
</section>
<section id="computational-methods" class="level3">
<h3 class="anchored" data-anchor-id="computational-methods">Computational Methods</h3>
<p><a href="https://cran.r-project.org/package=spconf">spconf</a> v1.0.1: Provides functions to compute the effective range of a smoothing matrix, which is a measure of the distance to which smoothing occurs. See <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0311440">Rainey and Keller (2024)</a> and <a href="https://academic.oup.com/jrsssa/article/183/3/1121/7056457?login=false">Keller and Szpiro (2020)</a> for algorithmic details and <a href="https://cran.r-project.org/web/packages/spconf/readme/README.html">README</a> for examples.</p>
</section>
<section id="data" class="level3">
<h3 class="anchored" data-anchor-id="data">Data</h3>
<p><a href="https://cran.r-project.org/package=PurpleAirAPI">PurpleAirAPI</a> v0.1.0: Provides tools for retrieving and analyzing air quality data from <a href="https://api.purpleair.com/">PurpleAir</a> sensors through an API. Functions enable downloading historical measurements, accessing sensor metadata, and managing API request limitations through chunked data retrieval.</p>
<p><a href="https://cran.r-project.org/package=sivirep">sivirep</a> v1.0.1: Provides tools for data wrangling, pre-processing, and generating automated reports from Colombia’s epidemiological surveillance system, <a href="https://portalsivigila.ins.gov.co/">SIVIGILA</a>, including a customizable R Markdown template for analysis and automatic generation of epidemiological reports. See the <a href="https://cran.r-project.org/web/packages/sivirep/vignettes/sivirep.html">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=STATcubeR">STATcubeR</a> v1.0.0: Implements an R client to import data from the STATcube REST API or via the open government data portal <a href="https://data.statistik.gv.at">Statistik Austria</a>. Look <a href="https://statistikat.github.io/STATcubeR/">here</a> and <a href="https://github.com/statistikat/STATcubeR">here</a> for more information. See the <a href="https://cran.r-project.org/web/packages/PurpleAirAPI/vignettes/getting-started-with-purpleairapi.html">vignette</a> to get started.</p>
<p><a href="https://cran.r-project.org/package=wbids">wbids</a> v0.1.0: Provides access to the World Bank’s <a href="https://datacatalog.worldbank.org/search/dataset/0038015">International Debt Statistics (IDS)</a> which contains creditor-debtor relationships between countries, regions, and institutions.</p>
<p><img src="wbids.png" class="img-fluid" alt="Schematic of data model"></p>
</section>
<section id="ecology" class="level3">
<h3 class="anchored" data-anchor-id="ecology">Ecology</h3>
<p><a href="https://cran.r-project.org/package=ecostate">ecostate</a> v0.2.0: Fits a state-space mass-balance model for marine ecosystems, which implements dynamics derived from <a href="https://ecopath.org/">Ecopath with Ecosim</a> while fitting to time series of fishery catch, biomass indices, age-composition samples, and weight-at-age data.</p>
<p><a href="https://cran.r-project.org/package=tidywater">tidywater</a> v0,6.2: Provides multiple water chemistry-based models and published empirical models in one standard format. Functions can be chained together to model a complete treatment process. See <a href="https://www.amazon.com/Water-Chemistry-Second-Mark-Benjamin/dp/147862308X">Benjamin (2002)</a>, <a href="https://onlinelibrary.wiley.com/doi/book/10.1002/9781118131473">Crittenden et al.&nbsp;(2012)</a> and <a href="https://www.epa.gov/sites/default/files/2017-03/documents/wtp_model_v._2.0_manual_508.pdf">US EPA Manual, (2001)</a> for background. There is an <a href="https://cran.r-project.org/web/packages/tidywater/vignettes/intro.html">Introduction</a> and two additional vignettes.</p>
<p><img src="tidywater.png" class="img-fluid" alt="Plot of ion concentrations"></p>
</section>
<section id="education" class="level3">
<h3 class="anchored" data-anchor-id="education">Education</h3>
<p><a href="https://cran.r-project.org/package=evolved">evolved</a> 1.0.0: Provides multiple tools to help educators (especially at the graduate level or in advanced undergraduate level courses) in general evolution classes. Tools include functions to simulate evolutionary processes (e.g., genetic drift, natural selection) and concepts (e.g.&nbsp;Hardy-Weinberg equilibrium, phylogenetic distribution of traits). There are eleven vignettes including <a href="https://cran.r-project.org/web/packages/evolved/vignettes/birthdeath_deeptime.html">birthdeath_deeptime</a> and <a href="https://cran.r-project.org/web/packages/evolved/vignettes/popgen_intro.html">popgen_intro</a>.</p>
<p><img src="evolved.png" class="img-fluid" alt="Photo of Bears"></p>
<p><a href="https://CRAN.R-project.org/package=exams2forms">exams2forms</a> v0.1-0: Implements automatic generation of quizzes or individual questions as (interactive) forms within <code>rmarkdown</code> or <code>quarto</code> documents based on <code>R/exams</code> exercises. See the <a href="https://cran.r-project.org/web/packages/exams2forms/vignettes/exams2forms.html">vignette</a>.</p>
<p><img src="exams.png" class="img-fluid" alt="Sample exercise"></p>
</section>
<section id="environment" class="level3">
<h3 class="anchored" data-anchor-id="environment">Environment</h3>
<p><a href="https://cran.r-project.org/package=OpeNoise">OpeNoise</a> v0.2-18: Provides functions to analyse, interpret and understand noise pollution data typically regular time series measured with sound meter. See <a href="https://dael.euracoustics.org/confs/landing_pages/fa2023/000063.html">Grasso, Masera and Scordino (2023)</a> for background and look <a href="https://github.com/Arpapiemonte/openoise-analysis/">here</a> for examples.</p>
<p><img src="OpeNoise.png" class="img-fluid" alt="Noise spectrogram"></p>
</section>
<section id="epidemiology" class="level3">
<h3 class="anchored" data-anchor-id="epidemiology">Epidemiology</h3>
<p><a href="https://cran.r-project.org/package=sivirep">sivirep</a> v1.0.1: Provides tools for data wrangling, pre-processing, and generating automated reports from Colombia’s epidemiological surveillance system, <a href="https://portalsivigila.ins.gov.co/">SIVIGILA</a> including a customizable R Markdown template for analysis and automatic generation of epidemiological reports. See the <a href="https://cran.r-project.org/web/packages/sivirep/vignettes/sivirep.html">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=vaccineff">vaccineff</a> v1.0.0: Provides tools for estimating vaccine effectiveness and related metrics and includes functions for managing key features and organizing cohort data, as well as estimating vaccine effectiveness. There are three vignettes including <a href="https://cran.r-project.org/web/packages/vaccineff/vignettes/cohort_design.html">Introduction to cohort design</a> and <a href="https://cran.r-project.org/web/packages/vaccineff/vignettes/vaccineff.html">vaccineff</a>.</p>
<p><img src="vaccineff.png" class="img-fluid" alt="Plot of vaccinestatus"></p>
</section>
<section id="genomics" class="level3">
<h3 class="anchored" data-anchor-id="genomics">Genomics</h3>
<p><a href="https://cran.r-project.org/package=divent">divent</a> v0.4-4: Implements a method to measure and partition diversity, including phylogenetic and functional diversity, with functions to estimate alpha, beta and gamma diversity of communities, See <a href="https://www.jstatsoft.org/article/view/v067i08">Marcon and Herault (2015)</a> for background and the <a href="https://cran.r-project.org/web/packages/divent/vignettes/divent.html">vignette</a> for an introduction.</p>
<p><img src="divent.png" class="img-fluid" alt="Plot f phylogenetic diversity by sample size"> <a href="https://cran.r-project.org/package=GRIN2">GRIN2</a> v1.0: Implements the GRIN algorithm as described in <a href="https://academic.oup.com/bioinformatics/article/29/17/2088/244947?login=false">Pounds et al.&nbsp;(2013)</a> for analyzing genomic lesion data and expands its analysis capabilities to answer additional scientific questions including a rigorous evaluation of the association of genomic lesions with RNA expression. See the <a href="https://cran.r-project.org/web/packages/GRIN2/vignettes/GRIN2.html">vignette</a>.</p>
<p><img src="GRIN2.png" class="img-fluid" alt="Annotated OncoPrint"></p>
<p><a href="https://cran.r-project.org/package=scregclust">scregclust</a> v0.2.0: Implements the scregclust algorithm described in <a href="https://www.nature.com/articles/s41467-024-53954-3">Larsson, Held, et al.&nbsp;(2024)</a> which reconstructs regulatory programs of target genes in scRNA-seq data. See the <a href="https://cran.r-project.org/web/packages/scregclust/vignettes/pbmc.html">vignette</a>.</p>
<p><img src="scregclust.png" class="img-fluid" alt="Plot of regulator network"></p>
</section>
<section id="health-technology-assessment" class="level3">
<h3 class="anchored" data-anchor-id="health-technology-assessment">Health Technology Assessment</h3>
<p><a href="https://cran.r-project.org/package=AIscreenR">AIscreenR</a> v0.1.1: Provides functions to conduct title and abstract screening in systematic reviews using large language models, such as the Generative Pre-trained Transformer (GPT) models from <a href="https://platform.openai.com/">OpenAI</a> and also tools for quality assessment of title and abstract screenings, as described in <a href="https://osf.io/preprints/osf/yrhzm">Vembye et al.&nbsp;(2024)</a>. See the <a href="https://cran.r-project.org/web/packages/AIscreenR/vignettes/Using-GPT-API-Models-For-Screening.html">vignette</a></p>
<p><a href="https://www.valueinhealthjournal.com/article/S1098-3015(12)01610-5/fulltext?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1098301512016105%3Fshowall%3Dtrue">maicplus</a> v0.1.1: Implements the method described in <a href="">Signorovitch et al (2012)</a> to facilitate matching adjusted indirect comparison (MAIC) analysis where the endpoint of interest is either time-to-event (e.g.&nbsp;overall survival) or binary (e.g.&nbsp;objective tumor response). There are seven vignettes including an <a href="https://cran.r-project.org/web/packages/maicplus/vignettes/introduction.html">Introduction</a> and <a href="https://cran.r-project.org/web/packages/maicplus/vignettes/anchored_survival.html">Anchored Survival Analysis</a>.</p>
<p><img src="maicplus.png" class="img-fluid" alt="Multiple Kaplan-Meier Plots"></p>
</section>
<section id="machine-learning" class="level3">
<h3 class="anchored" data-anchor-id="machine-learning">Machine Learning</h3>
<p><a href="https://cran.r-project.org/src/contrib/Archive/metasnf">metasnf</a> v1.1.2: Provides a framework for transforming multi-modal data into a wide range of similarity networks using the similarity network fusion (SNF) algorithm introduced by <a href="https://www.nature.com/articles/nmeth.2810">Wang et al.&nbsp;(2014)</a> and the meta clustering approach of <a href="https://ieeexplore.ieee.org/document/4053039">Caruana et al.&nbsp;(2006)</a>. Core functionality includes easy customization of distance metrics, clustering algorithms, and SNF hyperparameters. There are twenty-two vignettes including <a href="https://cran.r-project.org/web/packages/metasnf/vignettes/settings_matrix.html">The Settings Matrix</a> and <a href="https://cran.r-project.org/web/packages/metasnf/vignettes/a_complete_example.html">A Complete Example</a>.</p>
<p><img src="metasnf.png" class="img-fluid" alt="Plot of variable/solution associations"></p>
<p><a href="https://cran.r-project.org/package=ReSurv">ReSurv</a> V1.0.0: Implements Machine Learning models for predicting claim accounts using neural nets, extreme gradient boosting, and Cox model with splines to optimize the partial log-likelihood of proportional hazard models. There are five vignettes including an <a href="https://cran.r-project.org/web/packages/ReSurv/vignettes/Manuscript_replication_material.html">Introduction</a> and <a href="https://cran.r-project.org/web/packages/ReSurv/vignettes/cas_call.html">Claim Counts Prediction Using Individual Data</a>.</p>
<p><a href="https://cran.r-project.org/package=SVEMnet">SVEMnet</a> 1.2.1: Implements Self-Validated Ensemble Models (SVEM) <a href="https://www.sciencedirect.com/science/article/abs/pii/S0169743921002070?via%3Dihub">Lemkus et al.&nbsp;(2021)</a> using Elastic Net regression via <code>glmnet</code>, and also implements the the randomized permutation whole model test for SVEM <a href="https://www.sciencedirect.com/science/article/abs/pii/S0169743924000625?via%3Dihub">Karl (2024)</a>. See the <a href="https://cran.r-project.org/web/packages/SVEMnet/vignettes/SVEMnet_vignette.html">vignette</a>.</p>
<p><img src="SVEMnet.png" class="img-fluid" alt="Boxplots showing model test results"></p>
</section>
</div><div class="column" style="width:10%;">

</div><div class="column" style="width:45%;">
<section id="mathematics" class="level3">
<h3 class="anchored" data-anchor-id="mathematics">Mathematics</h3>
<p><a href="https://cran.r-project.org/package=cellularautomata">cellularautomata</a> v0.1.0: Provides functions to create cellular automata from <code>Wolfram</code> with plots and animations rules. The output of a cellular automaton is given as a matrix, making it easy to try to explore the possibility of predicting its time evolution using various statistical tools available in R. See Wolfram S. (2002, ISBN:1579550088) <em>A New Kind of Science</em> for background and the <a href="https://cran.r-project.org/web/packages/cellularautomata/vignettes/basic-usage.html">vignette</a> for examples.</p>
<p><img src="ca.gif" class="img-fluid" alt="Animation of Wolfram rule 193"></p>
<p><a href="https://CRAN.R-project.org/package=redcas">redcas</a> v0.1.1: Implements an interface to the <code>Lisp</code> based computer algebra system <a href="http://www.reduce-algebra.com">REDUCE</a> for interactively executing arbitrary <code>REDUCE</code> code interactively from <code>R</code>, returning output as character vectors. <code>R</code> code and <code>REDUCE</code> code can be interspersed. See <a href="https://cran.r-project.org/web/packages/redcas/readme/README.html">README</a> to get started.</p>
</section>
<section id="medicine" class="level3">
<h3 class="anchored" data-anchor-id="medicine">Medicine</h3>
<p><a href="https://cran.r-project.org/package=ClinTrialPredict">ClinTrialPredict</a> v0.0.4: Implements a method to predict the course of clinical trial with a time-to-event endpoint for both two-arm and single-arm designs. Simulation datasets can be generated based on the design settings. See the <a href="https://cran.r-project.org/web/packages/ClinTrialPredict/vignettes/ClinTrialPredict.pdf">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=Virusparies">Virusparies</a> v1.0.0: Implements a visualization tool for the <a href="https://github.com/lauberlab/VirusHunterGatherer">VirusHunterGather</a> computational pipeline for data-driven virus discovery from sequencing data from the NIH’s <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive</a> or the user’s own data. <a href="https://cran.r-project.org/web/packages/Virusparies/readme/README.html">README</a> points to eleven vignettes including <a href="https://github.com/SergejRuff/Virusparies?tab=readme-ov-file#virushuntergatherer-plot---vhgboxplot">VhgBoxplot</a> and <a href="https://github.com/SergejRuff/Virusparies?tab=readme-ov-file#virushuntergatherer-plot---vhgidentityscatterplot">VhgScatterPlot</a>.</p>
<p><img src="Virusparies.png" class="img-fluid" alt="Violinplot of Viral group by Contig Length"></p>
</section>
<section id="networks" class="level3">
<h3 class="anchored" data-anchor-id="networks">Networks</h3>
<p><a href="https://cran.r-project.org/package=ig.degree.betweenness">ig.degree.betweenness</a> v0.1.0: Implements the “Smith-Pittman” community detection algorithm for network analysis using <code>igraph</code> objects. This algorithm combines node degree and betweenness centrality measures to identify communities within networks, with a gradient evident in social partitioning. See <a href="https://arxiv.org/abs/2411.01394">Smith, Pittman and Xu (2024)</a> for background and look <a href="https://github.com/benyamindsmith/ig.degree.betweenness">here</a> for examples.</p>
<p><img src="ig.png" class="img-fluid" alt="Network diagram depicing clustering for UK faculty"></p>
</section>
<section id="pharma" class="level3">
<h3 class="anchored" data-anchor-id="pharma">Pharma</h3>
<p><a href="https://cran.r-project.org/package=autoslider.core">autoslider.core</a> v0.1.0: Provides functions to automatically create sides from <code>flextable</code> objects and create standard table templates often used in Study Results Endorsement Plan. It aims to save time, reduce errors and avoid the duplicated effort between statisticians and programmers. Look <a href="https://github.com/insightsengineering/autoslider.core">here</a> to see the code.</p>
</section>
<section id="statistics" class="level3">
<h3 class="anchored" data-anchor-id="statistics">Statistics</h3>
<p><a href="https://cran.r-project.org/package=BayesGP">BayesGP</a> v0.1.3: Implements Bayesian hierarchical models with flexible Gaussian process priors, focusing on extended latent Gaussian models and incorporating various Gaussian process priors for Bayesian smoothing. See <a href="https://journals.sagepub.com/doi/10.1177/09622802221134172">Zhang et al.&nbsp;(2023)</a>. <a href="https://www.tandfonline.com/doi/full/10.1080/10618600.2023.2289532">Zhang et al.&nbsp;(2024)</a>, and <a href="https://academic.oup.com/biometrics/article/77/3/785/7445030?login=false">Stringer, Brown, and Stafford (2021)</a> for the methods used. There are three vignettes including <a href="https://cran.r-project.org/web/packages/BayesGP/vignettes/BayesGP-Partial_Likelihood.html">Partial Likelihood</a> and <a href="https://cran.r-project.org/web/packages/BayesGP/vignettes/BayesGP-sGP.html">Fitting sGP</a>.</p>
<p><img src="BayesGP.png" class="img-fluid" alt="Plot of inferred function"></p>
<p><a href="https://cran.r-project.org/package=distfreereg">distfreereg</a> v1.0.1: Implements distribution-free goodness-of-fit regression testing for the mean structure of parametric models introduced in <a href="https://link.springer.com/article/10.1007/s10463-021-00786-3">Khmaladze (2021)</a>. There is an <a href="https://cran.r-project.org/web/packages/distfreereg/vignettes/v1_introduction.html">Introduction</a> and four additional vignettes including <a href="https://cran.r-project.org/web/packages/distfreereg/vignettes/v2_compare.html">Comparing Distributions</a>.</p>
<p><img src="distfree.png" class="img-fluid" alt="Estimated density plot"></p>
<p><a href="https://cran.r-project.org/package=GeoTox">GeoTox</a> v0.2.0: Implements a method to connect spatiotemporal exposure to individual and population-level risk via source-to-outcome continuum modeling. See <a href="https://www.medrxiv.org/content/10.1101/2024.09.23.24314096v2">Messier, Reif, and Marvel (2024)</a> and <a href="https://www.sciencedirect.com/science/article/pii/S0048969722060041?via%3Dihub">Eccles et al.&nbsp;(2023)</a> for the theory. There is an <a href="https://cran.r-project.org/web/packages/GeoTox/vignettes/introduction.html">Introduction</a> and a vignette on <a href="https://cran.r-project.org/web/packages/GeoTox/vignettes/package_data.html">Package Data</a>.</p>
<p><img src="GeoTox.png" class="img-fluid" alt="Spatial contraction maps"></p>
<p><a href="https://cran.r-project.org/package=OPSR">OPSR</a> v0.1.2: Implements ordinal probit switching regression models (a Heckman type selection model with an ordinal selection and continuous outcomes). See <a href="ordinal%20probit%20switching%20regression%20models%20-%20a%20Heckman%20type%20selection%20model%20with%20an%20ordinal%20selection%20and%20continuous%20outcomes.%20Different%20model%20specifications%20are%20allowed%20for%20each%20treatment/regime.%20For%20more%20details%20on%20the%20method,%20see%20Wang%20&amp;%20Mokhtarian%20(2024)%20%3Cdoi:10.1016/j.tra.2024.104072%3E%20or%20Chiburis%20&amp;%20Lokshin%20(2007)%20%3Cdoi:10.1177/1536867X0700700202%3E.">Wang &amp; Mokhtarian (2024)</a> and <a href="https://journals.sagepub.com/doi/10.1177/1536867X0700700202">Chiburis &amp; Lokshin (2007)</a> for details and <a href="https://cran.r-project.org/web/packages/OPSR/readme/README.html">README</a> for an example.</p>
</section>
<section id="time-series" class="level3">
<h3 class="anchored" data-anchor-id="time-series">Time Series</h3>
<p><a href="https://cran.r-project.org/package=tsmarch">tsmarch</a> v1.0.0: Implements feasible multivariate generalized autoregressive conditional heteroscedasticity (GARCH) models including dynamic conditional correlation (DCC), copula GARCH and generalized orthogonal GARCH with generalized hyperbolic distribution. See <a href="https://www.sciencedirect.com/science/article/abs/pii/S016971611930001X?via%3Dihub">Boudt et al.(2019)</a> for background and the vignettes <a href="https://cran.r-project.org/web/packages/tsmarch/vignettes/feasible_multivariate_garch.pdf">Feasible Multivariate Garch Models</a> and <a href="https://cran.r-project.org/web/packages/tsmarch/vignettes/tsmarch_demo.html">demo</a>.</p>
<p><img src="tsmarch.png" class="img-fluid" alt="Plot of Kurtosis with prediction"></p>
</section>
<section id="utilities" class="level3">
<h3 class="anchored" data-anchor-id="utilities">Utilities</h3>
<p><a href="https://cran.r-project.org/package=deident">deident</a> v1.0.0: Implements a framework for the replicable removal of personally identifiable data (PID) in data sets and provides methods to suit different data types based on the suggestions of <a href="https://nvlpubs.nist.gov/nistpubs/ir/2015/NIST.IR.8053.pdf">Garfinkel (2015)</a> and the <a href="https://ico.org.uk/media/1061/anonymisation-code.pdf">ICO “Guidelines on Anonymization” (2012)</a>. There are nine vignettes including <a href="https://cran.r-project.org/web/packages/deident/vignettes/rationale.html">Rational for De-identification</a> and <a href="https://cran.r-project.org/web/packages/deident/vignettes/worked_example.html">Worked Example</a>.</p>
<p><a href="https://cran.r-project.org/package=lofifonts">lofifonts</a> v0.1.3: Provides tools for alternate font rendering with bitmap and vector fonts that allow for custom layout and rendering using pixel coordinates and line drawing. There are five vignettes including <a href="https://cran.r-project.org/web/packages/lofifonts/vignettes/bitmap-fonts.html">Using bitmap fonts</a> and <a href="https://cran.r-project.org/web/packages/lofifonts/vignettes/unicode-support.html">Unicode Support</a>.</p>
<p><img src="lofifonts.png" class="img-fluid" alt="Sample bitmap font"> <a href="https://cran.r-project.org/package=surveydown">surveydown</a> v0.5.1: Provides functions to generate surveys using markdown and R code chunks. Features include controls for conditional skip logic (skip to a page based on an answer to a question), conditional display logic (display a question based on an answer to a question), a customizable progress bar, and a wide variety of question types, including multiple choice (single choice and multiple choices), select, text, numeric, multiple choice buttons. Designers can also leverage the reactive capabilities of <code>shiny</code> to create dynamic and interactive surveys. See the <a href="https://cran.r-project.org/web/packages/surveydown/vignettes/surveydown.html">vignette</a>.</p>
</section>
<section id="visualization" class="level3">
<h3 class="anchored" data-anchor-id="visualization">Visualization</h3>
<p><a href="https://cran.r-project.org/package=effectplots">effectplots</a> v0.2.0: Implementation various effect plots useful for regression and probabilistic classification tasks including partial dependence plots <a href="https://projecteuclid.org/journals/annals-of-statistics/volume-29/issue-5/Greedy-function-approximation-A-gradient-boosting-machine/10.1214/aos/1013203451.full">Friedman (2021)</a>, accumulated local effect plots and M-plots from <a href="https://academic.oup.com/jrsssb/article/82/4/1059/7056085?login=false">Apley and Zhu (2016)</a>, and plots that describe the statistical associations between model response and features. Look <a href="https://github.com/mayer79/effectplots">here</a> for examples.</p>
<p><img src="effectplots.svg" class="img-fluid" alt="Plots of traininv vs test data"> <a href="https://cran.r-project.org/package=PlotFTIR">plotFTIR</a> v1.0.0: Implements journal-quality Fourier Transform Infra-Red (FTIR) spectral plots using <code>ggplot2</code>. See the <a href="https://cran.r-project.org/web/packages/PlotFTIR/vignettes/plotting_ftir_spectra.html">vignette</a>.</p>
<p><img src="plotFTIR.png" class="img-fluid" alt="FTIR Spectra plot"></p>
<p><a href="https://cran.r-project.org/package=plotthis">plotthis</a> v0.4.0: Extends <code>ggplot2</code> to provide a high-level API with wide range of options to create publication-quality plots effortlessly. It is particularly useful for creating complex plots with multiple layers, facets, and annotations. Look <a href="https://pwwang.github.io/plotthis/">here</a> for examples.</p>
<p><img src="plotthis.png" class="img-fluid" alt="Correlation pairs plots"></p>
<p><a href="https://cran.r-project.org/package=RegrCoeffsExplorer">RegrCoeffsExplorer</a> v1.1.0: Implements a visualization tool that offers a nuanced understanding of regression dynamics, going beyond traditional per-unit interpretation of continuous and categorical variables th highlight the impact of unit changes as well as larger shifts such as interquartile changes. See the vignettes <a href="https://cran.r-project.org/web/packages/RegrCoeffsExplorer/vignettes/BetaVisualizer.html">BetaVisualizer</a> and <a href="https://cran.r-project.org/web/packages/RegrCoeffsExplorer/vignettes/OddsRatioVisualizer.html">OddsRatioVisualizer</a>.</p>
<p><img src="RegCoef.png" class="img-fluid" alt="Boxblor and OR plot sshing effect sizes"> <a href="https://cran.r-project.org/package=scatterbar">scatterbar</a> v0.1.0: Implements a new tool for visualizing proportional data across spatially resolved contexts by combining concepts of scatter plots and stacked bar charts. <code>scatterbar</code> plots allow users to create scattered bar chart plots, which effectively display the proportions of different categories at each (x, y) location. See <a href="https://www.biorxiv.org/content/10.1101/2024.08.14.606810v2">Velazquez et al.&nbsp;(2024)</a> for details. There are five vignettes including <a href="https://cran.r-project.org/web/packages/scatterbar/vignettes/getting-started-with-scatterbars.html">Getting Started</a> and <a href="https://cran.r-project.org/web/packages/scatterbar/vignettes/customizing_scatterbar.html">Customizing scatterbar</a>.</p>
<p><img src="scatterbar.png" class="img-fluid" alt="Example of scatterbarchart"> <a href="https://cran.r-project.org/package=talkr">talkr</a> v0.1.3: Provides visualization, analysis and quality control tools for gaining insights into the nature of conversational data. See <a href="https://aclanthology.org/2022.acl-long.385/">Dingemanse et al., (2022)</a> for details and the <a href="https://cran.r-project.org/web/packages/talkr/vignettes/workflows.html">vignette</a> for examples.</p>
<p><img src="talkr.png" class="img-fluid" alt="Conversation patterns"></p>
<p><img src=".png" class="img-fluid" alt=""></p>
</section>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>