<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2024_August_T40</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="2024AugustT40_files/libs/clipboard/clipboard.min.js"></script>
<script src="2024AugustT40_files/libs/quarto-html/quarto.js"></script>
<script src="2024AugustT40_files/libs/quarto-html/popper.min.js"></script>
<script src="2024AugustT40_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="2024AugustT40_files/libs/quarto-html/anchor.min.js"></script>
<link href="2024AugustT40_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2024AugustT40_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="2024AugustT40_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="2024AugustT40_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="2024AugustT40_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">2024_August_T40</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>“Top 40” is back, broadcasting on this new R Consortium blog. I hope to continue to monthly evaluation of R packages that ran for several years on RStudio’s <a href="https://rviews.rstudio.com/">R Views Blog</a>. The following is an idiosyncratic selection of the forty best new R packages to submitted to CRAN in August 2024 organized into fourteen categories: Artificial Intelligence, Computational Methods, Data, Ecology, Environment, Genomics, Machine Learning, Medicine, Pharma, Science, Statistics, Time Series, Utilities, and Visualization.</p>
<section id="artificial-intelligence" class="level3">
<h3 class="anchored" data-anchor-id="artificial-intelligence">Artificial Intelligence</h3>
<p><a href="https://cran.r-project.org/package=gemini.R">gemini.R</a> v0.5.2: Provides an R interface to <a href="https://ai.google.dev/api?lang=python">Google Gemini API</a> for advanced language processing, text generation, and other AI-driven capabilities within the R environment. See <a href="https://cran.r-project.org/web/packages/gemini.R/readme/README.html">README</a> to get started.</p>
<p><a href="https://cran.r-project.org/package=promptr">promptr</a> v1.0.0: Provides functions to form and submit prompts to OpenAI’s Large Language Models. Designed to be particularly useful for text classification problems in the social sciences. See <a href="https://joeornstein.github.io/publications/ornstein-blasingame-truscott.pdf">Ornstein, Blasingame, &amp; Truscott (2024)</a> for details and <a href="https://cran.r-project.org/web/packages/promptr/readme/README.html">README</a> for an example.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/promptr.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Scatter plot: GPT-3.5 computed sentiment vs.&nbsp;hand coded sentiment</figcaption>
</figure>
</div>
</section>
<section id="computational-methods" class="level3">
<h3 class="anchored" data-anchor-id="computational-methods">Computational Methods</h3>
<p><a href="https://cran.r-project.org/package=qvirus">qvirus</a> v 0.0.2: Provides code and resources to explore the intersection of quantum computing and artificial intelligence (AI) in the context of analyzing Cluster of Differentiation 4 (CD4) lymphocytes and optimizing antiretroviral therapy (ART) for human immunodeficiency virus (HIV). See the vignettes <a href="https://cran.r-project.org/web/packages/qvirus/vignettes/Introduction.html">Introduction</a>, <a href="https://cran.r-project.org/web/packages/qvirus/vignettes/Applications.html">Applications</a>, and <a href="https://cran.r-project.org/web/packages/qvirus/vignettes/Entanglement.html">Entanglement</a>.</p>
<p><a href="https://cran.r-project.org/package=RcppBessel">RcppBessel</a> v1.0.0: Exports an <code>Rcpp</code> interface for the Bessel functions in the ‘Bessel’ package, which can then be called from the <code>C++</code> code of other packages. For the original ‘Fortran’ implementation of these functions see <a href="https://dl.acm.org/doi/10.1145/212066.212078">Amos (1995)</a>. There is a <a href="https://cran.r-project.org/web/packages/RcppBessel/vignettes/RcppBessel.html">vignette</a>.</p>
</section>
<section id="data" class="level3">
<h3 class="anchored" data-anchor-id="data">Data</h3>
<p><a href="https://cran.r-project.org/package=aebdata">aebdata</a> v0.1.0: Facilitates access to the data from the <a href="https://www.ipea.gov.br/atlasestado/">Atlas do Estado Brasileiro</a> maintained by the Instituto de Pesquisa Econômica Aplicada (Ipea). It allows users to search for specific series, list series or themes, and download data when available. See the <a href="https://cran.r-project.org/web/packages/aebdata/vignettes/exploring-data.html">vignette</a>.</p>
<p><a href="https://CRAN.R-project.org/package=capesData">capesData</a> v0.0.1: Provides information on activities to promote scholarships in Brazil and abroad for international mobility programs recorded in the <a href="https://dadosabertos.capes.gov.br/dataset/">CAPES database</a> from 2010 to 2019. See <a href="https://cran.r-project.org/web/packages/capesData/readme/README.html">README</a> to get started.</p>
</section>
<section id="ecology" class="level3">
<h3 class="anchored" data-anchor-id="ecology">Ecology</h3>
<p><a href="https://cran.r-project.org/package=priorCON">priorCON</a> v0.1.1: Provides a tool-set that incorporates graph community detection methods into systematic conservation planning. It is designed to enhance spatial prioritization by focusing on the protection of areas with high ecological connectivity and on clusters of features that exhibit strong ecological linkages. See the <a href="https://cran.r-project.org/web/packages/priorCON/vignettes/Introduction.html">Introduction</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/priorCON.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Flowchart of priorCon analysis.</figcaption>
</figure>
</div>
</section>
<section id="environment" class="level3">
<h3 class="anchored" data-anchor-id="environment">Environment</h3>
<p><a href="https://CRAN.R-project.org/package=prior3D">prior3D</a> v0.1.0: Offers a comprehensive tool set for 3D systematic conservation planning, conducting nested prioritization analyses across multiple depth levels and ensuring efficient resource allocation throughout the water column. See <a href="https://onlinelibrary.wiley.com/doi/10.1111/gcb.16268">Doxa et al.&nbsp;(2022)</a> for background, and the <a href="https://cran.r-project.org/web/packages/prior3D/vignettes/Introduction.html">vignette</a> for an example</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/prior3D.jpeg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Flow chart for 3D Prioritization Analysis</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=raem">raem</a> v0.1.0: Implements a model of single-layer groundwater flow in steady-state under the Dupuit-Forchheimer assumption can be created by placing elements such as wells, area-sinks and line-sinks at arbitrary locations in the flow field. See <a href="https://www.sciencedirect.com/book/9780123165503/analytic-element-modeling-of-groundwater-flow?via=ihub=">Haitjema (1995)</a> for the underlying theory, and the vignettes <a href="https://cran.r-project.org/web/packages/raem/vignettes/vign-01-overview.html">Overview</a> and <a href="https://cran.r-project.org/web/packages/raem/vignettes/vign-02-export-spatial.html">Exporting spatial data</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/raem.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Contour plot of well head with streamlines</figcaption>
</figure>
</div>
</section>
<section id="genomics" class="level3">
<h3 class="anchored" data-anchor-id="genomics">Genomics</h3>
<p><a href="https://CRAN.R-project.org/package=kmeRtone">kmeRtone</a> v 1.0: Provides functions for multi-purpose k-meric enrichment analysis which measures the enrichment of k-mers by comparing the population of k-mers in the case loci with an internal negative control group consisting of k-mers from regions close to, yet sufficiently distant from, the case loci. This method captures both the local sequencing variations and broader sequence influences, while also correcting for potential biases. See the <a href="https://github.com/SahakyanLab/kmeRtone">GitHub repo</a> for an overview.</p>
<p><a href="https://cran.r-project.org/package=rYWAASB">rYWAASB</a> v0.1: Provides new ranking algorithm to distinguish the top-ranked genotypes. “WAASB” refers to the “Weighted Average of Absolute Scores” provided by <a href="https://acsess.onlinelibrary.wiley.com/doi/10.2134/agronj2019.03.0220">Olivoto et al.&nbsp;(2019)</a> which quantifies the stability of genotypes across different environments using linear mixed-effect models. See the <a href="https://cran.r-project.org/web/packages/rYWAASB/vignettes/rYWAASB_manual.html">vignette</a> for an example.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/WAASB.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">WAASB Biplot</figcaption>
</figure>
</div>
</section>
<section id="machine-learning" class="level3">
<h3 class="anchored" data-anchor-id="machine-learning">Machine Learning</h3>
<p><a href="https://cran.r-project.org/package=cvLM">cvLM</a> v1.0.4: Provides efficient implementations of cross-validation techniques for linear and ridge regression models, leveraging <code>C++</code> code with <code>Rcpp</code> that supports leave-one-out, generalized, and K-fold cross-validation methods. See <a href="https://cran.r-project.org/web/packages/cvLM/readme/README.html">README</a> for an example.</p>
<p><a href="https://cran.r-project.org/package=geodl">geodl</a> v0.2.0: Provides tools for semantic segmentation of geospatial data using convolutional neural network-based deep learning, including utility functions for manipulating data, model checks, functions to implement a UNet architecture with 4 blocks in the encoder, assessment metrics and more. The package relies on <a href="http://torch.ch/">torch</a> but does not require installing a Python environment. Models can be trained using a Compute Unified Device Architecture (CUDA)-enabled graphics processing unit (GPU). There are ten vignettes including <a href="https://cran.r-project.org/web/packages/geodl/vignettes/spatialPredictionDemo.html">spatialPredictionDemo</a> and <a href="https://cran.r-project.org/web/packages/geodl/vignettes/topoDLDemo.html">topoDLDemo</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/geodl.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Plot of topo chips examples</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=idiolect">idiolect</a> v1.0.1: Provides functions for the comparative authorship analysis of disputed and undisputed texts within the Likelihood Ratio Framework for expressing evidence in forensic science, and implements well-known algorithms including <a href="https://www.tandfonline.com/doi/abs/10.1080/09296174.2011.533591">Smith and Aldridge’s (2011)</a> Cosine Delta, and <a href="https://asistdl.onlinelibrary.wiley.com/doi/10.1002/asi.22954">Koppel and Winter’s (2014)</a> Impostors Method. See the <a href="https://cran.r-project.org/web/packages/idiolect/vignettes/idiolect.htm">vignette</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/idiolect.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Plots of Score densities</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=kdml">kdml</a> v1.0.0: Implements distance metrics for mixed-type data consisting of continuous, nominal, and ordinal variables which can be used in any distance-based algorithm, such as distance-based clustering. See <a href="https://arxiv.org/abs/2306.01890">Ghashti and Thompson (2024)</a> for <code>dkps()</code> methodology, <a href="https://open.library.ubc.ca/soa/cIRcle/collections/ubctheses/24/items/1.0443975">Ghashti (2024)</a> for <code>dkss()</code> methodology, and the <a href="https://cran.r-project.org/web/packages/kdml/vignettes/kdml.html%20for%20an%20example">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=kerntools">kerntools</a> v1.0.2: Provides kernel functions for diverse types of data including, but not restricted to: non negative and real vectors, real matrices, categorical and ordinal variables, sets, strings, plus other utilities like kernel similarity, kernel Principal Components Analysis (PCA) and features’ importance for Support Vector Machines (SVMs). See the <a href="https://cran.r-project.org/web/packages/kerntools/vignettes/kerntools.html">vignette</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/kerntools.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Scatter plot for Drac kernal PCA</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=MorphoRegions">MorphoRegions</a> v0.1.0: Provides functions to computationally identify regions in serially homologous structures such as, but not limited to, the vertebrate backbone. Regions are modeled as segmented linear regressions with each segment corresponding to a region, and region boundaries (or breakpoints) corresponding to changes along the serially homologous structure.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/Morpho.gif" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Animation of process of serially identifying regions</figcaption>
</figure>
</div>
</section>
<section id="medicine" class="level3">
<h3 class="anchored" data-anchor-id="medicine">Medicine</h3>
<p><a href="https://CRAN.R-project.org/package=neuroUp">neuroUP</a> v0.3.1: Provides functions to calculate the precision in mean differences (raw or Cohen’s D) and correlation coefficients for different sample sizes using permutations of the collected functional magnetic resonance imaging (fMRI) data. See <a href="https://osf.io/preprints/psyarxiv/cz32t">Klapwijk et al.&nbsp;(2024)</a> for background, and the <a href="https://cran.r-project.org/web/packages/neuroUp/vignettes/neuroUp.html">vignette</a> for and introduction.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/neuroUp.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Barplot of proportion permutations</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=smiles">smiles</a> v0.1-0: Provides tools aimed at making data synthesis and evidence evaluation easier for both experienced practitioners and newcomers. See the <a href="https://onlinelibrary.wiley.com/doi/10.1002/9781119536604.ch22">Cochrane Handbook for Systematic Reviews of Interventions</a>, and the <a href="https://cran.r-project.org/web/packages/smiles/vignettes/An_overview_on_pakcage_smiles.html">vignette</a> for examples.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/smiles.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Plots of trials sequence analysis for treatment vs.&nbsp;placebo</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=tsgc">tsgc</a> v0.0: Provides tools to analyze and forecast epidemic trajectories based on a dynamic Gompertz model and a state space approach that uses the Kalman filter for robust estimation of the non-linear growth pattern commonly observed in epidemic data. See <a href="https://hdsr.mitpress.mit.edu/pub/ozgjx0yn/release/4">Harvey and Kattuman (2020)</a>, <a href="https://royalsocietypublishing.org/doi/10.1098/rsif.2021.0179">Harvey and Kattuman (2021)</a> and <a href="https://www.jbs.cam.ac.uk/wp-content/uploads/2024/03/cchle-tsgc-paper-2024.pdf">Ashby et al.&nbsp;(2024)</a> for background and the <a href="https://cran.r-project.org/web/packages/tsgc/vignettes/tsgc_vignette.html">vignette</a> for details.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/tsgc.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Time series of moving averages for Covid-19 model</figcaption>
</figure>
</div>
</section>
<section id="pharma" class="level3">
<h3 class="anchored" data-anchor-id="pharma">Pharma</h3>
<p><a href="https://CRAN.R-project.org/package=admiralpeds">admiralpeds</a> v0.1.0: Provides a toolbox for programming Clinical Data Standards Interchange Consortium <a href="https://www.cdisc.org/standards/foundational/adam">(CDISC)</a> compliant Analysis Data Model (ADaM) data sets in R. See the <a href="https://cran.r-project.org/web/packages/admiralpeds/vignettes/advs.html">vignette</a> for an example.</p>
<p><a href="https://cran.r-project.org/package=MALDIcellassay">MALDIcellassay</a> v0.4.47: Implements tools to conduct automated cell-based assays using Matrix-Assisted Laser Desorption/Ionization (MALDI) methods for high-throughput screening of signals responsive to treatments. The methodologies were introduced by <a href="https://www.nature.com/articles/s41598-018-29677-z">Weigt et al.&nbsp;(2018)</a> and <a href="https://www.nature.com/articles/s41596-021-00624-z">Unger et al.&nbsp;(2021)</a>. See the <a href="https://cran.r-project.org/web/packages/MALDIcellassay/vignettes/Example.html">vignette</a> for an example.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/MALDI.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Spectral Plot</figcaption>
</figure>
</div>
</section>
<section id="science" class="level3">
<h3 class="anchored" data-anchor-id="science">Science</h3>
<p><a href="https://cran.r-project.org/package=barrks">barrks</a> v1.0.0: Implements models to calculate the bark beetle phenology and their submodels, onset of infestation, beetle development, diapause initiation, mortality, which can be customized and combined. Models include PHENIPS-Clim, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0378112707004057?via%3Dihub">PHENIPS</a>, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0304380019302832?via%3Dihub">RITY</a>, <a href="https://www.sciencedirect.com/science/article/pii/S0304380020302088?via%3Dihub">CHAPY</a>, and <a href="https://onlinelibrary.wiley.com/doi/10.1111/gcb.14766">BSO</a>. There are five vignettes including <a href="https://cran.r-project.org/web/packages/barrks/vignettes/bso.html">The BSO model</a> and <a href="https://cran.r-project.org/web/packages/barrks/vignettes/model-comparison.html">Example: Model Comparison</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/barrks.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Plots of generations calculated by RITY</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=fluxible">fluxible</a> v0.0.1: Provides functions to process the raw data from closed loop flux chamber (or tent) setups into ecosystem gas fluxes usable for analysis. Implemented models include include exponential <a href="https://www.sciencedirect.com/science/article/pii/S0168192318302880?via%3Dihub">Zhao et al.&nbsp;(2018)</a> and quadratic and linear models to estimate the fluxes from the raw data. See the <a href="https://cran.r-project.org/web/packages/fluxible/vignettes/fluxible_practical_example.html">vignette</a> for an example.</p>
</section>
<section id="statistics" class="level3">
<h3 class="anchored" data-anchor-id="statistics">Statistics</h3>
<p><a href="https://cran.r-project.org/package=bage">bage</a> v0.7.4: Provides functions for Bayesian estimation and forecasting of age-specific rates, probabilities, and means, based on the <a href="https://kaskr.github.io/adcomp/Introduction.html">Template Model Builder</a>. There are six vignettes including the <a href="https://cran.r-project.org/web/packages/bage/vignettes/vig2_math.html">Mathematical Details</a> and an <a href="https://cran.r-project.org/web/packages/bage/vignettes/vig6_mortality.html">Example</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/bage.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Scatter plots estimating mortality rates</figcaption>
</figure>
</div>
<p><a href="https://CRAN.R-project.org/package=ClustMC">clustMC</a> v0.1.1: implements cluster-based multiple comparisons tests and also provides a visual representation in the form of a dendrogram. See <a href="https://www.jstor.org/stable/1400690">Rienzo, Guzmán &amp; Casanoves (2002)</a> and <a href="https://www.jstor.org/stable/1400402?origin=crossref">Bautista, Smith &amp; Steiner (1997)</a> and the <a href="https://cran.r-project.org/web/packages/ClustMC/vignettes/CustomPlots.html">vignette</a> for examples.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/clustMC.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Dendogram plot</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=svycdiff">svycdiff</a> v0.1.1: Provides three methods for estimating the population average controlled difference for a given outcome between levels of a binary treatment, exposure, or other group membership variables of interest for clustered, stratified survey samples where sample selection depends on the comparison group. See <a href="https://arxiv.org/abs/2406.19597">Salerno et al.&nbsp;(2024)</a> for background and the <a href="https://cran.r-project.org/web/packages/svycdiff/vignettes/svycdiff.html">vignette</a> for an example.</p>
<p><a href="https://cran.r-project.org/package=wishmom">wishmom</a> v1.1.0: Provides functions for computing moments and coefficients related to the Beta-Wishart and Inverse Beta-Wishart distributions, including functions for calculating the expectation of matrix-valued functions of the Beta-Wishart distribution, coefficient matrices, and expectation of matrix-valued functions of the inverse Beta-Wishart distribution. See the <a href="https://cran.r-project.org/web/packages/wishmom/vignettes/wishmom_vignettes.html">vignete</a> for details.</p>
</section>
<section id="time-series" class="level3">
<h3 class="anchored" data-anchor-id="time-series">Time Series</h3>
<p><a href="https://cran.r-project.org/web/packages/tican/index.html">tican</a> v1.0.1: Provides functions to analyze and plot time-intensity curves such as those that arise from contrast-enhanced ultrasound images. See the <a href="https://cran.r-project.org/web/packages/tican/vignettes/tican-vignette.html">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=tsgc">tidychangepoint</a> v0.0.1: Provides a tidy, unified interface for several different changepoint detection algorithms along with a consistent numerical and graphical reporting leveraging the <code>broom</code> and <code>ggplot2</code> packages. See the <a href="https://cran.r-project.org/web/packages/tidychangepoint/vignettes/tidychangepoint.html">vignette</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/tidychange.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Time series with change points</figcaption>
</figure>
</div>
</section>
<section id="utilities" class="level3">
<h3 class="anchored" data-anchor-id="utilities">Utilities</h3>
<p><a href="https://cran.r-project.org/package=fio">fio</a> v0.1.2: Provides tools to simplify the process of importing and managing input-output matrices from Microsoft <code>Excel</code> into R. It leverages the <code>R6</code> class for memory-efficient object-oriented programming implements all linear algebra computations in <code>Rust</code>. See the <a href="https://cran.r-project.org/web/packages/fio/vignettes/getting_started.html">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=litedown">litedown</a> v0.2: Implements a lightweight version of R Markdown which enables rendering R Markdown to Markdown without using <code>knitr</code>, and Markdown to lightweight <code>HTML/LaTeX</code> documents using the <code>commonmark</code> package instead of <code>Pandoc</code>. This package can be viewed as a trimmed-down version of R Markdown and <code>knitr</code> which does not aim at rich Markdown features or a large variety of output formats. There are vignettes on <a href="https://cran.r-project.org/web/packages/litedown/vignettes/markdown-examples.html">Markdown Examples</a>, <a href="https://cran.r-project.org/web/packages/litedown/vignettes/markdown-output.html">HTML Output Examples</a>, and <a href="https://cran.r-project.org/web/packages/litedown/vignettes/slides.html">Making HTMLSlides</a>.</p>
<p><a href="https://cran.r-project.org/package=maestro">maestro</a> v0.2.0: Implements a framework for creating and orchestrating data pipelines allowing users to organize, orchestrate, and monitor multiple pipelines in a single project. There are four vignettes including a <a href="https://cran.r-project.org/web/packages/maestro/vignettes/maestro-1-quick-start.html">Quick Start</a> Guide and <a href="https://cran.r-project.org/web/packages/maestro/vignettes/maestro-3-use-cases.html">Use Cases</a>.</p>
<p><a href="https://cran.r-project.org/package=osum">osum</a> v0.1.0: Inspired by <code>S-PLUS</code> function <code>objects.summary()</code>, provides a function that returns data class, storage mode, mode, type, dimension, and size information for R objects in the specified environment. Various filtering and sorting options are also proposed. See the <a href="https://cran.r-project.org/web/packages/osum/vignettes/osum.html">vignette</a>.</p>
<p><a href="https://cran.r-project.org/package=overtureR">overtureR</a> v0.2.3: Implements n integrated R interface to the <a href="https://docs.overturemaps.org/">Overture’ API</a> which allows R users to return Overture data as <code>dbplyr</code> data frames or materialized <code>sf</code> spatial data frames. See <a href="https://cran.r-project.org/web/packages/overtureR/readme/README.html">README</a> for examples.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/overtureR.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Overture map of Broadway</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=RcppMagicEnum">RcppMagicEnum</a> v0.0.1; Provides <code>Rcpp</code> bindings to header-only modern <code>C++</code> template library <a href="https://github.com/Neargye/magic_enum">Magic Enum</a>. See <a href="https://cran.r-project.org/web/packages/RcppMagicEnum/readme/README.html">README</a> to get started.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/tidymodelr.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">PCA Graph</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=tidymodlr">tidymodelr</a> v1.0.0: Provides a function to transform long data into a matrix form to allow for ease of input into modelling packages for regression, principal components, imputation or machine learning along with level analysis wrapper functions for correlation and principal components analysis. See <a href="https://cran.r-project.org/web/packages/tidymodlr/readme/README.html">README</a> for examples.</p>
</section>
<section id="visualization" class="level2">
<h2 class="anchored" data-anchor-id="visualization">Visualization</h2>
<p><a href="https://cran.r-project.org/package=bullseye">bullseye</a> v0.1.0: Provides a tidy data structure and visualizations for multiple or grouped variable correlations, general association measures scagnostics and other pairwise scores suitable for numerical, ordinal and nominal variables. Supported measures include distance correlation, maximal information, ace correlation, Kendall’s tau, and polychoric correlation. There are three vignettes including <a href="https://cran.r-project.org/web/packages/bullseye/vignettes/calc_pairwise.html">Calculating Pairwise Scores</a> and <a href="https://cran.r-project.org/web/packages/bullseye/vignettes/vis_pairwise.html">Visualizing Pairwise Scores</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/bullseye.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Visualizing pairwise scores</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=flowmapper">flowmapper</a> v0.1.2: Adds flow maps to <code>ggplot2</code> plots. These are layers which visualize the nodes as circles and the bilateral flows between the nodes as bidirectional half-arrows. Look <a href="https://github.com/JohMast/flowmapper">here</a> for details and examples.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/flowmapper.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Flow map</figcaption>
</figure>
</div>
<p><a href="https://cran.r-project.org/package=ggreveal">ggreveal</a> v0.1.3: Provides functions that make it easy to reveal <code>ggplot2</code> graphs incrementally. The functions take a plot produced with <code>ggplot2</code> and return a list of plots showing data incrementally by panels, layers, groups, the values in an axis or any arbitrary aesthetic. See the <a href="https://github.com/weverthonmachado/ggreveal">GitHub repo</a> for examples.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../#1/ggreveal.gif" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Revealed layers of ggplot</figcaption>
</figure>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>