<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2024_September_T40</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="2024SeptTop40_files/libs/clipboard/clipboard.min.js"></script>
<script src="2024SeptTop40_files/libs/quarto-html/quarto.js"></script>
<script src="2024SeptTop40_files/libs/quarto-html/popper.min.js"></script>
<script src="2024SeptTop40_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="2024SeptTop40_files/libs/quarto-html/anchor.min.js"></script>
<link href="2024SeptTop40_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2024SeptTop40_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="2024SeptTop40_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="2024SeptTop40_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="2024SeptTop40_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">2024_September_T40</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="ai" class="level3">
<h3 class="anchored" data-anchor-id="ai">AI</h3>
<p><a href="https://cran.r-project.org/package=groqR">groqR</a> v0.0.1: Provides a suite of functions and <code>RStudio</code> Add-ins leveraging the capabilities of open-source Large Language Models (LLMs) to support R developers. Features include text rewriting, translation, and general query capabilities. Programming-focused functions provide assistance with debugging, translating, commenting, documenting, and unit testing code, as well as suggesting variable and function names. Look <a href="https://github.com/GabrielKaiserQFin/groqR">here</a> for examples.</p>
</section>
<section id="archaeology" class="level3">
<h3 class="anchored" data-anchor-id="archaeology">Archaeology</h3>
<p><a href="https://cran.r-project.org/package=eratosthenes">eratosthenes</a> v0.0.2: Estimates unknown historical or archaeological dates subject to relationships with other dates and absolute constraints, derived as marginal densities from the full joint conditional distribution. Includes rule-based estimation of the production dates of artifact types. See <a href="https://volweb.utk.edu/~scolli46/sceGUTChronology.pdf">Collins-Elliott (2024)</a> for background and the vignettes on <a href="https://cran.r-project.org/web/packages/eratosthenes/vignettes/aligning-relative-sequences.html">aligning relative sequences</a> and <a href="https://cran.r-project.org/web/packages/eratosthenes/vignettes/gibbs-sampling-for-archaeological-dates.html">gibbs sampling for archaeology dates</a>.</p>
</section>
<section id="biology" class="level3">
<h3 class="anchored" data-anchor-id="biology">Biology</h3>
<p><a href="https://cran.r-project.org/package=pcvr">pcvr</a> v1.0.0: Provides functions to analyse common types of plant phenotyping data and asimplified interface to longitudinal growth modeling and select Bayesian statistics. See <a href="https://journals.sagepub.com/doi/10.1177/2515245918771304">Kruschke (2018)</a>, <a href="https://psycnet.apa.org/doiLanding?doi=10.1037%2Fa0029146">Kruschke (2013)</a> and <a href="https://www.nature.com/articles/s41562-021-01177-7">Kruschke (2021)</a> for background on the Bayesian methods. There are four vignettes including <a href="https://cran.r-project.org/web/packages/pcvr/vignettes/bellwether.Rmd">Bellwether workflow</a> and <a href="https://cran.r-project.org/web/packages/pcvr/vignettes/longitudinal.html">Longitudinal Growth Modeling</a>.</p>
<p><img src="../#2/pcvr.png" class="img-fluid" alt="Plot of water use efficiency by genotype"></p>
<p><a href="https://cran.r-project.org/package=orthGS">orthGS</a> v0.1.5: Provides tools to analyze and infer orthology and paralogy relationships between glutamine synthetase proteins in seed plants. See the vignettes <a href="https://cran.r-project.org/web/packages/orthGS/vignettes/Searching_for_Orthologous.html">Searching for Orthologous</a> and <a href="https://cran.r-project.org/web/packages/orthGS/vignettes/Unraveling_the_hiden_paralogous.html">Unraveling the Hiden Paralogous</a>.</p>
<p><img src="../#2/orthGS.png" class="img-fluid" alt="Plot of species and protein tree"></p>
</section>
<section id="computational-methods" class="level3">
<h3 class="anchored" data-anchor-id="computational-methods">Computational Methods</h3>
<p><a href="https://cran.r-project.org/package=XDNUTS">XDNUTS</a> v1.2: Implements Hamiltonian Monte Carlo for both continuous and discontinuous posterior distributions with customisable trajectory length termination criterion. See <a href="https://academic.oup.com/biomet/article/107/2/365/5799014?login=false">Nishimura et al.&nbsp;(2020)</a> for the original Discontinuous Hamiltonian Monte Carlo; and <a href="https://arxiv.org/abs/1111.4246">Hoffman et al.&nbsp;(2014)</a> and <a href="https://arxiv.org/abs/1601.00225">Betancourt (2016)</a> possible Hamiltonian Monte Carlo termination criteria. The <a href="https://cran.r-project.org/web/packages/XDNUTS/vignettes/Intro_to_XDNUTS.html">vignette</a> offers examples.</p>
<p><img src="../#2/XDNUTS.png" class="img-fluid" alt="Plots examining marginal densities"></p>
</section>
<section id="data" class="level3">
<h3 class="anchored" data-anchor-id="data">Data</h3>
<p><a href="https://cran.r-project.org/package=clintrialx">clintrialx</a> v0.1.0: Provides functions to fetch clinical trial data from sources like [ClinicalTrials.gov](https://clinicaltrials.gov/} and the Clinical Trials <a href="https://aact.ctti-clinicaltrials.org/">Access to Aggregate Content</a> database that supports pagination and bulk downloads. See the <a href="https://cran.r-project.org/web/packages/clintrialx/vignettes/usecase-doc.html">vignette</a>.</p>
<p><img src="../#2/clintrialsx.png" class="img-fluid" alt="Histogram of the number of studies in each status category"></p>
<p><a href="https://cran.r-project.org/package=ColOpenData">ColOpenData</a> v0.3.0: Provides tools to download and wrangle Colombian socioeconomic, geospatial,population and climate data from <a href="https://www.dane.gov.co/">DANE</a> at the National Administrative Department of Statistics and <a href="https://ideam.gov.co">IDEAM</a> at the Institute of Hydrology, Meteorology and Environmental Studies. It solves the problem of Colombian data being issued in different web pages and sources by using functions that allow the user to select the desired database and download it without having to do the exhausting acquisition process. There are six vignettes including <a href="https://cran.r-project.org/web/packages/ColOpenData/vignettes/climate_data.html">How to download climate data</a> and <a href="https://cran.r-project.org/web/packages/ColOpenData/vignettes/population_projections.html">Population Projections</a>.</p>
<p><img src="../#2/ColOpenData.png" class="img-fluid" alt="Map of Columbia showing Internet Coverage."></p>
<p><a href="https://cran.r-project.org/package=modgo">modgo</a> v1.0.1: Provides functions to generate synthetic data from a real dataset using the combination of rank normal inverse transformation with the calculation of correlation matrix and completely artificial data may be generated through the use of <a href="https://www.taylorfrancis.com/books/mono/10.1201/9781420038040/fitting-statistical-distributions-zaven-karian-edward-dudewicz">Generalized Lambda Distribution</a> and Generalized Poisson Distribution. See the <a href="https://cran.r-project.org/web/packages/modgo/vignettes/modgo_vignette.html">vignette</a>.</p>
<p><img src="../#2/modgo.png" class="img-fluid" alt="Correlation matrices for simulated and real data"> ### Ecology</p>
<p><a href="https://cran.r-project.org/package=douconca">douconca</a> v1.2.1: Implements the two step double constrained correspondence analysis (dc-CA) for analyzing multi-trait multi-environment ecological data described in<a href="https://link.springer.com/article/10.1007/s10651-017-0395-x">ter Braak et al.&nbsp;(2018)</a>. This algorithm combines and extends community or sample and species-level analyses.</p>
<p><img src="../#2/douconca.png" class="img-fluid" alt="Plots showing impact of environmental variables"></p>
<p><a href="https://cran.r-project.org/web/packages/GeoThinneR/index.html#:~:text=https%3A//CRAN.R%2Dproject.org/package%3DGeoThinneR">GeoThinneR</a> v1.1.0: Provides efficient geospatial thinning algorithms to reduce the density of coordinate data while maintaining spatial relationships. Implements K-D Tree and brute-force distance-based thinning, as well as grid-based and precision-based thinning methods. See <a href="https://hdl.handle.net/10446/86202">Elseberg et al.&nbsp;(2012)</a> for background and the <a href="https://cran.r-project.org/web/packages/GeoThinneR/vignettes/GeoThinneR.html">vignette</a> for examples.</p>
<p><img src="../#2/GeoThinneR.png" class="img-fluid" alt="Simulated species data by longitude and lattitude"></p>
</section>
<section id="genomics" class="level3">
<h3 class="anchored" data-anchor-id="genomics">Genomics</h3>
<p><a href="https://cran.r-project.org/package=easybio">easybio</a> v1.1.0: Provides a toolkit for single-cell annotation with the <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC9825416/">CellMarker2.0</a> database and streamlines biological label assignment in single-cell RNA-seq data and facilitates transcriptomic analysis, including preparation of <a href="https://portal.gdc.cancer.gov/">TCGA</a> and <a href="https://www.ncbi.nlm.nih.gov/geo/">GEO</a> datasets, differential expression analysis and visualization of enrichment analysis results. See <a href="https://www.biorxiv.org/content/10.1101/2024.09.14.609619v1">Wei Cui (2024)</a> for details and the two vignettes <a href="https://cran.r-project.org/web/packages/easybio/vignettes/example-bulk-rna-seq-workflow.html">bulk RNAsewuence workflow</a> and <a href="https://cran.r-project.org/web/packages/easybio/vignettes/example-single-cell-annotation.html">Single Cell Annotation</a> for examples.</p>
<p><img src="../#2/easybio.png" class="img-fluid" alt="Plots showing criteria for differential expression"></p>
<p><a href="https://cran.r-project.org/package=GenoPop">GenoPop</a> v0.9.3: Implements tools for efficient processing of large, whole genome genotype data sets in variant call format including several functions to calculate commonly used population genomic metrics and a method for reference panel free genotype imputation. See <a href="https://www.authorea.com/users/825581/articles/1221193-genopop-impute-efficient-and-accurate-whole-genome-genotype-imputation-in-non-model-species-for-evolutionary-genomic-research">Gurke &amp; Mayer (2024)</a> for background and the <a href="https://cran.r-project.org/web/packages/GenoPop/GenoPop.pdf">vignette</a> to get started. <img src="../#2/GenoPop.png" class="img-fluid" alt="Plots of gene populations"></p>
<p><a href="https://cran.r-project.org/package=SuperCell">SuperCell</a> v1.0: Provides tools to aggregate large single-cell data into metacell dataset by merging together gene expression of very similar cells See the vignettes <a href="https://cran.r-project.org/web/packages/SuperCell/vignettes/a_SuperCell.html">Example of SuperCell pipeline</a> and <a href="https://cran.r-project.org/web/packages/SuperCell/vignettes/b_Combined_vs_independent_sample_processing.html">SuperCell runs for different samples</a>.</p>
<p><img src="../#2/SuperCell.png" class="img-fluid" alt="Plots of metacells clusters"></p>
</section>
<section id="lingusitics" class="level3">
<h3 class="anchored" data-anchor-id="lingusitics">Lingusitics</h3>
<p><a href="https://cran.r-project.org/package=maxent.ot">maxent.ot</a> v1.0.0: Provides tools to fit Maximum Entropy models to phnology data. See <a href="https://sites.socsci.uci.edu/~cjmayer/papers/cmayer_et_al_maxent_ot_accepted.pdf">Mayer, Tan &amp; Zuraw</a> and the <a href="https://cran.r-project.org/web/packages/maxent.ot/vignettes/maxent-ot_vignette.html">vignette</a> for an overview.</p>
</section>
<section id="machine-learning" class="level3">
<h3 class="anchored" data-anchor-id="machine-learning">Machine Learning</h3>
<p><a href="https://cran.r-project.org/package=dsld">dsld</a> v0.2.2: Provides statistical and graphical tools for detecting and measuring discrimination and bias, be it racial, gender, age or other. Detection and remediation of bias in machine learning algorithms. See the <a href="https://cran.r-project.org/web/packages/dsld/vignettes/Quick_Start.html">Quick Start Guide</a>.</p>
</section>
<section id="medicine" class="level3">
<h3 class="anchored" data-anchor-id="medicine">Medicine</h3>
<p><a href="https://cran.r-project.org/package=SurvMA">SurvMA</a> v1.6.8: Implements a model averaging approach to predict personalized survival probabilities by using a weighted average of multiple candidate models to approximate the conditional survival function.Two scenarios of candidate models are allowed: the partial linear Cox model and the time-varying coefficient Cox model. See <a href="https://www.sciencedirect.com/science/article/abs/pii/S0167947323000701?via%3Dihub">Wang (2023)</a> for details and look <a href="https://github.com/Stat-WangXG/SurvMA">here</a> for an example.</p>
<p><img src="../#2/SurvMA.png" class="img-fluid" alt="Schmatic of model fitting process"></p>
<p><a href="https://cran.r-project.org/package=wintime">wintime</a> v0.2.0: Provides methods to perform an analysis of time-to-event clinical trial data using various methods that calculate and compare treatment effects on ordered composite endpoints. See <a href="https://pubmed.ncbi.nlm.nih.gov/38417455/">Troendle et al.&nbsp;(2024)</a> for the details of the methods and the <a href="https://cran.r-project.org/web/packages/wintime/vignettes/wintime_vignette.html">vignette</a> for examples.</p>
</section>
<section id="pharma" class="level3">
<h3 class="anchored" data-anchor-id="pharma">Pharma</h3>
<p><a href="https://cran.r-project.org/package=sdtm.oak">sdtm.oak</a> v0.1.0: Provides a framework to develop CDISC, <a href="https://www.cdisc.org/standards/foundational/sdtmig">SDTM</a> datasets in R and potentially automate the process. There are six vignettes including on on <a href="https://cran.r-project.org/web/packages/sdtm.oak/vignettes/algorithms.html">Algorithms</a>.</p>
<p><img src="../#2/sdtm.jpeg" class="img-fluid" alt="Schematic of algorithms"></p>
</section>
<section id="statistics" class="level3">
<h3 class="anchored" data-anchor-id="statistics">Statistics</h3>
<p><a href="https://cran.r-project.org/package=dpasurv">dpasurv</a> v0.1.0: Provides functions to implement dynamic path analysis for survival data via Aalen’s additive hazards model. See <a href="https://link.springer.com/article/10.1007/s10985-006-9004-2">Fosen et al., (2006)</a> for details. There is an <a href="https://cran.r-project.org/web/packages/dpasurv/vignettes/dpa.Rmd">oveview</a> and a <a href="https://cran.r-project.org/web/packages/dpasurv/vignettes/ggplot_demo.html">vignette</a> on plotting with <code>ggplot2</code>.</p>
<p><img src="../#2/dpasurv.png" class="img-fluid" alt="Plot showing mediator variable over time by dose response."></p>
<p><a href="https://cran.r-project.org/package=LearnVizLMM">LearnVizLLM</a> v1.0.0: Implements tools to summarize the characteristics of linear fixed models without data or a fitted model by converting code code for fitting <code>nlme::lme()</code> and <code>lme4::lmer()</code> models into tables, equations, and visuals. See the <a href="https://cran.r-project.org/web/packages/LearnVizLMM/vignettes/LearnVizLMM.html">vignette</a> for details.</p>
<p><img src="../#2/LearnVizLLM.png" class="img-fluid" alt="Schematic of mutilevel, nested model"></p>
<p><a href="https://cran.r-project.org/package=lnmixsurv">lnmixsurv</a> v3.1.6: Combines the mixture distributions of <a href="https://link.springer.com/article/10.1007/s11336-009-9121-4">Fruhwirth-Schnatter(2006)</a> and the data augmentation techniques of <a href="https://www.tandfonline.com/doi/abs/10.1080/01621459.1987.10478458">Tanner and Wong (1987)</a> to implement Bayesian Survival models that accommodate different behaviour over time and consider higher censored survival times. There are five vignettes including a [Get started guide}</p>
</section>
<section id="utility" class="level3">
<h3 class="anchored" data-anchor-id="utility">Utility</h3>
<p><a href="https://CRAN.R-project.org/package=fctutils">fctutils</a> v0.0.7: Provides a collection of utility functions for manipulating and analyzing factor vectors in R. It offers tools for filtering, splitting, combining, and reordering factor levels based on various criteria. See the <a href="https://cran.r-project.org/web/packages/fctutils/vignettes/fctutils.html">vignette</a>.</p>
<p><a href="">pikchr</a>V0.97 : Provides an interface to <a href="https://pikchr.org">pikchr</a> q markup language for creating diagrams within technical documentation. See the <a href="https://cran.r-project.org/web/packages/pikchr/vignettes/Examples.html">vignette</a> for examples.</p>
<p><img src="../#2/pitchar.png" class="img-fluid" alt="Example block diagram"></p>
<p><a href="https://cran.r-project.org/package=rix">rnix</a> v0.12.4: Provides tools to run the <a href="https://en.wikipedia.org/wiki/Nix_(package_manager)">nix package manager</a>. There are fifteen vignettes including a <a href="https://cran.r-project.org/web/packages/rix/vignettes/a-getting-started.html">Getting Started Gude</a>.</p>
<p><a href="https://CRAN.R-project.org/package=qs2">qs2</a> v0.1.1: Provides tools to efficiently serialize <code>R</code> objects using one of two compression formats: the <code>qs2</code> format, which uses R serialization while optimizing compression and disk I/O, and the <code>qdata</code> format which uses custom serialization to achieve slightly faster performance and better compression. <code>qs2</code> format can be directly converted to the standard <code>RDS</code>. See the <a href="https://cran.r-project.org/web/packages/qs2/vignettes/vignette.rmd">vignette</a></p>
</section>
<section id="visualization" class="level3">
<h3 class="anchored" data-anchor-id="visualization">Visualization</h3>
<p><a href="https://cran.r-project.org/package=ggalign">ggalign</a> v0.0.4: Implements an extension to <code>ggplot2</code> that offers various tools for organizing and arranging plots including the ability to consistently align a specific axis across multiple <code>ggplot</code> objects. There are seven vignettes including <a href="https://cran.r-project.org/web/packages/ggalign/vignettes/complete-examples.html">Examples</a> and <a href="https://cran.r-project.org/web/packages/ggalign/vignettes/heatmap-layout.html">Heatmap Layout</a>.</p>
<p><img src="../#2/ggalign.png" class="img-fluid" alt="Heatmap with bar chart"></p>
<p><a href="https://CRAN.R-project.org/package=sfcurve">sfcurv</a> v1.0: Implements all possible forms of 2x2 and 3x3 space-filling curves, i.e., the generalized forms of the <a href="https://en.wikipedia.org/wiki/Hilbert_curve">Hilbert curve</a>, the <a href="https://en.wikipedia.org/wiki/Peano_curve">Peano curve</a> and the Peano curve in the <a href="https://eudml.org/doc/141086">meander type</a>. Look <a href="https://jokergoo.github.io/sfcurve/">here</a> for ec=xamples.</p>
<p><img src="../#2/sfcurve.png" class="img-fluid" alt="Plot of 8 space filling curves"></p>
<p><a href="https://cran.r-project.org/package=surreal">surreal</a> v0.0.1: Implements the <em>Residual (Sur)Realism</em> algorithm described by <a href="">Stefanski (2007)</a> to generate datasets that reveal hidden images or messages in their residual plots. See <a href="https://cran.r-project.org/web/packages/surreal/readme/README.html">README</a> for examples.</p>
<p><img src="../#2/surreal.png" class="img-fluid" alt="R logohiddenina residual plot"> <a href="https://cran.r-project.org/package=survSAKK">survSAKK</a> v1.3.1: Provides functions to incorporate various statistics and layout customization options to enhance the efficiency and adaptability of the Kaplan-Meier plots. See the <a href="https://cran.r-project.org/web/packages/survSAKK/vignettes/surv.plot.html">vignette</a>.</p>
<p><img src="../#2/survSAKK.png" class="img-fluid" alt="Kaplan-Meier curve with risk table"></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>